ci-automation/vendor-testing/qemu.sh

```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
+# Copyright (c) 2021 The Flatcar Maintainers.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+
+set -euo pipefail
+
+# Test execution script for the qemu vendor image.
+# This script is supposed to run in the SDK container.
+
+work_dir="$1"; shift
+arch="$1"; shift
+vernum="$1"; shift
+tapfile="$1"; shift
+
+# $@ now contains tests / test patterns to run
+
+source ci-automation/ci_automation_common.sh
+
+mkdir -p "${work_dir}"
+cd "${work_dir}"
+
+echo "++++ QEMU test: downloading ${QEMU_IMAGE_NAME} for ${vernum} (${arch}) ++++"
+copy_from_buildcache "images/${arch}/${vernum}/${QEMU_IMAGE_NAME}" .
```

Comment:
I'm not super fond of the test script reaching out to the buildcache directly, the path to the image should be passed from the outside.

Reply:
It's configurable: https://github.com/flatcar-linux/scripts/blob/main/ci-automation/ci-config.env
This can be changed e.g. by using a separate branch, or overridden by CI (change the default in the env file before calling the script).
In a custom set-up the build cache could e.g. be on the local machine.
However I don't expect this parameter to change often, so I'd like to persist a sane default somewhere so CI is not required to feed this information.

Reply:
I'm fine with persisting the default, but I'm saying I would suggest to separate out "fetching images" from "executing tests". Since test_run() fetches "artifacts" the image fetching can also happen in test.sh. That way "qemu.sh" only has to test a provided image. Think of it this way: instead of passing credentials to buildcache all the way down, we do privileged stuff early, and then we don't need as many privileges deeper in the call stack.

Reply:
I think it's also worth considering to pass the URL and the test run can decide to download locally or pass it to kola as done with Digital Ocean where the API takes a public URL